core/transaction_files/transaction.py

```diff
@@ -0,0 +1,148 @@
+import time as _time
+from dataclasses import dataclass
+from dataclasses_json import dataclass_json
+import binascii
+import collections
+import json
+from ..wallet_files.wallet import Wallet
+from Crypto.PublicKey import RSA
+from Crypto.Random import get_random_bytes
+from Crypto.Signature import pkcs1_15
+from Crypto.Hash import SHA, SHA256
+
+# unneccessary class - should be removed
+@dataclass
+class TransactionData:
+    def __init__(self):
+        recipient: str
+        coin: str
+        value: str
+        
+# neccessary class should be expanded upon
+@dataclass
+class TransactionInput:
+    transaction_data:str
+    output_index: int
+    transaction_hash:str
+    public_key: bytes
+    signature: bytes = None
+    
+    def _to_json(self)-> str:
+        if self.signature == None:
+            return json.dumps({
+                "transaction_data": self.transaction_data,
+                "output_index": self.output_index,
+                "transaction_hash": self.transaction_hash
+            })
+        return json.dumps({
+            "transaction_data": self.transaction_data,
+            "output_index": self.output_index,
+            "transaction_hash": self.transaction_hash,
+            "public_key": self.public_key,
+            "signature": self.signature
+        })
+@dataclass 
+class TransactionOutput:
+    amount: int
+    coin_type: str
+    public_key_hash: str
+    
+    def _to_json(self)-> str:
+        return self.to_json()
+    
+@dataclass
+class TransactionNew:
+    sender: Wallet# this is now the wallet address(could possibly change to wallet object)
+    inputs: list[TransactionInput]
+    outputs: list [TransactionOutput]
+    time: float = _time.time()
+
+    # TODO: what is the identity property?
+    def _to_dict(self) -> collections.OrderedDict:
+        return collections.OrderedDict(
+            {
+                "sender": str(self.sender.address),
+                "inputs": [transaction_input._to_json() for transaction_input in self.inputs],
+                "outputs": [transaction_output._to_json() for transaction_output in self.outputs],
+                "time": str(self.time),
+            }
+        )
+    def to__dict(self):
```

Comment:
Do we need to make this a separate function or does the regular dict method not work